src/Projekt/Args.fs

```diff
@@ -1,111 +1,90 @@
-    | None -> V4_5
-    | _ -> failwith "invalid framework version argument specified"
+            | _ -> failwith "not given a full path"
 
-let private parseDirection s =
-    match s with
-    | ToLower "up" -> Up
-    | ToLower "down" -> Down
-    | _ -> failwith "invalid direction specified"
+    [<Verb("reference", HelpText = "reference a dll")>]
+    type public ReferenceOptions = 
+        {   [<Value(0, Required = true, MetaName = "project path")>]    projectPath : string
+            [<Value(1, Required = true, MetaName = "reference path")>]  referencePath : string  }
+    with 
+        member x.ToOperation =
+            match x.projectPath, x.referencePath with
+            | FullPath project, FullPath reference ->
+                {ProjPath = project; Reference = reference} |> Reference
+            | _,_ -> failwith "one or both paths were invalid"
 
-let private parser = UnionArgParser.Create<Args>()
-
-let private (|Options|) (args : string list) =
-    let results = parser.Parse(List.toArray args)
-    results
-
-let (|FullPath|_|) (path : string) =
-    try 
-        Path.GetFullPath path |> Some
+    [<Verb("movefile", HelpText = "Move a file within a project")>]
+    type public MoveFileOptions = 
+        {   [<Value(0, Required = true, MetaName = "project path")>]    projectPath : string
+            [<Value(1, Required = true, MetaName = "file path")>]       filePath : string
+            [<Option(Required = true)>]                                 direction : string
+            [<Option(Default = 1)>]                                     repeat : int    }
     with
-    | _ -> None
-
-let commandUsage = "projekt (init|reference|movefile|addfile|delfile|version) /path/to/project [/path/to/(file|project)]"
+        member x.ToOperation =
+            match x.projectPath, x.filePath, Direction.Create x.direction with
+            | FullPath project, FullPath file, dir -> 
+                {ProjPath = project; FilePath = file; Direction = dir; Repeat = x.repeat} |> MoveFile
+            | _,_,_ -> failwith "invalid paths or direction"
 
-let parse (ToList args) : Result<Operation> =
-    try
-        match args with
-        | [] -> 
-            parser.Usage commandUsage 
-            |> Failure
+    [<Verb("addfile", HelpText = "Add a file to a project")>]
+    type public AddFileOptions = 
+        {   [<Value(0, Required = true, MetaName = "project path")>]    projectPath : string
+            [<Value(1, Required = true, MetaName = "file path")>]       filePath : string
+            [<Option>]                                                  link : string option
+            [<Option(Default = true)>]                                  compile : bool  }
+    with 
+        member x.ToOperation =
+            match x.projectPath, x.filePath with
+            | FullPath project, FullPath file -> 
+                {ProjPath = project; FilePath = file; Link = x.link; Compile = x.compile} |> AddFile 
```

Comment:
this line can be shortened by putting each record field assignment on a new line.